Makefile

```diff
@@ -1,40 +1,63 @@
-VERSION := 8.0.5
-
-# soname - libcs50.so.<major_version>
-SONAME := libcs50.so.$(shell echo $(VERSION) | head -c 1)
+VERSION := 8.0.6
+MAJOR_VERSION := $(shell echo $(VERSION) | head -c 1)
 
 # installation directory (/usr/local by default)
 DESTDIR ?= /usr/local
+MANDIR ?= share/man/man3
+
+SRC := src/cs50.c
+INCLUDE := src/cs50.h
+DOCS := $(wildcard docs/*.adoc)
+MANS := $(addprefix debian/, $(DOCS:%.adoc=%.3)) 
+
+CFLAGS=-Wall -Wextra -Werror -pedantic -std=c99
+
+OS := $(shell uname)
+# Linux
+ifeq ($(OS),Linux)
+	LIB_BASE := libcs50.so
+	LIB_MAJOR := libcs50.so.$(MAJOR_VERSION)
+	LIB_VERSION := libcs50.so.$(VERSION)
+	LINKER_FLAGS := -Wl,-soname,$(LIB_MAJOR)
+# Mac
+else ifeq ($(OS),Darwin)
+	LIB_BASE := libcs50.dylib
+	LIB_MAJOR := libcs50-$(MAJOR_VERSION).dylib
+	LIB_VERSION := libcs50-$(VERSION).dylib
+	LINKER_FLAGS := -Wl,-install_name,$(LIB_MAJOR)
+endif
 
-.PHONY: build
-build: clean
-	$(CC) -c -fPIC -std=c99 -Wall -o cs50.o src/cs50.c
-	$(CC) -shared -Wl,-soname,$(SONAME) -o libcs50.so.$(VERSION) cs50.o
-	rm -f cs50.o
-	ln -s libcs50.so.$(VERSION) $(SONAME)
-	ln -s $(SONAME) libcs50.so
-	install -D -m 644 src/cs50.h build/include/cs50.h
+LIBS := $(addprefix build/lib/, $(LIB_BASE) $(LIB_MAJOR) $(LIB_VERSION))
+
+.PHONY: all
+all: $(LIBS) $(MANS)
+
+$(LIBS): $(SRC) $(INCLUDE) Makefile
+	$(CC) $(CFLAGS) -fPIC -shared $(LINKER_FLAGS) -o $(LIB_VERSION) $(SRC)
+	ln -s $(LIB_VERSION) $(LIB_MAJOR)
+	ln -s $(LIB_MAJOR) $(LIB_BASE)
+	mkdir -p src build/include
+	install -m 644 src/cs50.h build/include/cs50.h
 	mkdir -p build/lib build/src/libcs50
-	mv libcs50.so* build/lib
-	cp -r src/* build/src/libcs50
+	mv $(LIB_VERSION) $(LIB_MAJOR) $(LIB_BASE) build/lib
+	cp -r $(SRC) $(INCLUDE) build/src/libcs50
 
 .PHONY: install
-install: build docs
-	mkdir -p $(DESTDIR) $(DESTDIR)/share/man/man3
+install: all
+	mkdir -p $(DESTDIR) $(DESTDIR)/$(MANDIR)
 	cp -r build/* $(DESTDIR)
-	cp -r debian/docs/* $(DESTDIR)/share/man/man3
+	cp debian/docs/* $(DESTDIR)/$(MANDIR)
```

Comment:
@kzidane are you asking about the Linux `export LD_LIBRARY_PATH=/usr/local/lib` command asked for on line 35 of the updated README.md? 

As per [Apple Dynamic Libraries](https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/UsingDynamicLibraries.html) documentation, `/usr/local` is automatically checked for libraries. 

A user can install elsewhere as desired, setting `LD_LIBRARY_PATH`.